soc/nxp/imxrt/imxrt7xx/hifi4/soc.c

```diff
@@ -0,0 +1,148 @@
+	 * - IRQ 26 (SEL 21): EDMA1-6
+	 * - IRQ 27 (SEL 22): EDMA1-7
+	 * - IRQ 28 (SEL 23): MICFIL
+	 * - IRQ 29 (SEL 24): SAI0
+	 * - IRQ 30 (SEL 25): SAI1
+	 * - IRQ 31 (SEL 26): SAI2
+	 */
+	INPUTMUX_Init(INPUTMUX0);
+
+	INPUTMUX_AttachSignal(INPUTMUX0, 1, kINPUTMUX_Flexcomm0ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 2, kINPUTMUX_Flexcomm2ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 3, kINPUTMUX_Wdt1ToDspInterrupt);
+
+	INPUTMUX_AttachSignal(INPUTMUX0, 4, kINPUTMUX_Gpio0Irq0ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 5, kINPUTMUX_Gpio0Irq1ToDspInterrupt);
+
+	INPUTMUX_AttachSignal(INPUTMUX0, 6, kINPUTMUX_GpioInt2ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 7, kINPUTMUX_GpioInt3ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 8, kINPUTMUX_GpioInt4ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 9, kINPUTMUX_GpioInt5ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 10, kINPUTMUX_GpioInt6ToDspInterrupt);
+
+	INPUTMUX_AttachSignal(INPUTMUX0, 11, kINPUTMUX_GpioInt7ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 12, kINPUTMUX_Spi14ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 13, kINPUTMUX_Mu2AToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 14, kINPUTMUX_Mu4BToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 15, kINPUTMUX_Dma1Irq0ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 16, kINPUTMUX_Dma1Irq1ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 17, kINPUTMUX_Dma1Irq2ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 18, kINPUTMUX_Dma1Irq3ToDspInterrupt);
+
+	INPUTMUX_AttachSignal(INPUTMUX0, 19, kINPUTMUX_Dma1Irq4ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 20, kINPUTMUX_Dma1Irq5ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 21, kINPUTMUX_Dma1Irq6ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 22, kINPUTMUX_Dma1Irq7ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 23, kINPUTMUX_MicfilToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 24, kINPUTMUX_Sai0ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 25, kINPUTMUX_Sai1ToDspInterrupt);
+	INPUTMUX_AttachSignal(INPUTMUX0, 26, kINPUTMUX_Sai2ToDspInterrupt);
+
+	INPUTMUX_Deinit(INPUTMUX0);
+}
+
+__weak void mimxrt798s_hifi4_clock_init(void)
+{
+	CLOCK_AttachClk(kOSC_CLK_to_FCCLK0);
+	CLOCK_SetClkDiv(kCLOCK_DivFcclk0Clk, 1U);
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm0), okay)
+	SET_UP_FLEXCOMM_CLOCK(0);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(flexcomm2), okay)
+	SET_UP_FLEXCOMM_CLOCK(2);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(edma1), okay)
+	CLOCK_EnableClock(kCLOCK_Dma1);
+	RESET_ClearPeripheralReset(kDMA1_RST_SHIFT_RSTn);
+	edma_enable_all_request(1);
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(sai0), okay)
+	/* SAI clock 368.64 / 15 = 24.576MHz */
+	CLOCK_AttachClk(kAUDIO_PLL_PFD3_to_AUDIO_VDD2);
+	CLOCK_AttachClk(kAUDIO_VDD2_to_SAI012);
+	CLOCK_SetClkDiv(kCLOCK_DivSai012Clk, 15U);
+	RESET_ClearPeripheralReset(kSAI0_RST_SHIFT_RSTn);
+#endif
+}
+
+void soc_early_init_hook(void)
```

Comment:
Please ensure the appropriate `Kconfig`  is enabled. Else this will not get invoked